tests/searchcommands/test_internals_v2.py

```diff
@@ -257,76 +263,6 @@ def test_record_writer_with_random_data(self, save_recording=False):
         # P2 [ ] TODO: Verify that RecordWriter gives consumers the ability to finish early by calling
         # RecordWriter.flush(finish=True).
 
-        if save_recording:
-
-            cls = self.__class__
-            method = cls.test_record_writer_with_recordings
-            base_path = os.path.join(self._recordings_path, '.'.join((cls.__name__, method.__name__, six.text_type(time()))))
-
-            with gzip.open(base_path + '.input.gz', 'wb') as f:
-                pickle.dump(test_data, f)
-
-            with open(base_path + '.output', 'wb') as f:
-                f.write(writer._ofile.getvalue())
-
-        return
-
-    def test_record_writer_with_recordings(self):
-
-        cls = self.__class__
-        method = cls.test_record_writer_with_recordings
-        base_path = os.path.join(self._recordings_path, '.'.join((cls.__name__, method.__name__)))
-
-        for input_file in iglob(base_path + '*.input.gz'):
-
-            with gzip.open(input_file, 'rb') as ifile:
-                test_data = pickle.load(ifile)
-
-            writer = RecordWriterV2(StringIO(), maxresultrows=10)  # small for the purposes of this unit test
-            write_record = writer.write_record
-            fieldnames = test_data['fieldnames']
-
-            for values in test_data['values']:
-                record = OrderedDict(izip(fieldnames, values))
-                try:
-                    write_record(record)
-                except Exception as error:
-                    self.fail(error)
-
-            for message_type, message_text in test_data['messages']:
-                writer.write_message(message_type, '{}', message_text)
-
-            for name, metric in six.iteritems(test_data['metrics']):
-                writer.write_metric(name, metric)
-
-            writer.flush(finished=True)
-
-            # Read expected data
-
-            expected_path = os.path.splitext(os.path.splitext(input_file)[0])[0] + '.output'
-
-            with io.open(expected_path, 'rb') as ifile:
-                expected = ifile.read()
-
-            expected = self._load_chunks(StringIO(expected))
-
-            # Read observed data
-
-            ifile = writer._ofile
-            ifile.seek(0)
-
-            observed = self._load_chunks(ifile)
-
-            # Write observed data (as an aid to diagnostics)
-
-            observed_path = expected_path + '.observed'
-            observed_value = ifile.getvalue()
-
-            with io.open(observed_path, 'wb') as ifile:
-                ifile.write(observed_value)
-
-            self._compare_chunks(observed, expected)
```

Comment:
I see from your commit msg that this was removed due to portability issues, I'm curious to know more though not a blocker for the PR.

Reply:
So here's the basic problem.

> We have this test function called test_record_writer_with_recordings: https://github.com/splunk/splunk-sdk-python/blob/master/tests/searchcommands/test_internals_v2.py#L274
> 
> It appears to me that this is essentially test_record_writer_with_random_data except with static test fixtures: https://github.com/splunk/splunk-sdk-python/blob/master/tests/searchcommands/test_internals_v2.py#L174
> 
> Why I think this: I loaded it up in Python 2.x and it ends up building a dictionary with a fieldnames value of` OrderedDict([(u'fieldnames', [u'_serial', u'_time', u'random_bytes', u'random_dict', u'random_integers', u'random_unicode'])` which implies to me that it’s similar data as in the random_data version of the test.
> 
> Why do I say “in Python 2.x”? Well.
> 
> It seems as if the “recordings” are pickled data, which I take to mean that it’s an object that’s been serialized into a file.
> 
> The problem is, when you serialize internal object structures there’s no guarantee that they’re portable across versions.
> 
> So of course if you try to un-pickle in Python 3, the data — which I assume was originally serialized in Python 2 — is invalid.
>
> BUT the recordings seem to just be the same kind of stuff you’d get in random_data, just less random.
>
> But you could just reproduce that by feeding in the same Python seed. So I kind of feel like what if we just delete this function entirely

After working with @shakeelmohamed we came to the conclusion that this test scenario is fundamentally redundant anyway (as it can be easily regenerated via passing in the correct seed), so I cut it.